### PR TITLE
Study view performance

### PR DESCRIPTION
**Pull request by aderidder**
Some small performance improvements for the study view:
* deletion of the survival plots
* tables only update when they're visible
* tables show loading icon when filtering

@zhx828 are you willing to review?

### CODE REVIEW

**Comment by zhx828** [comment]
@pieterlukasse No problem.

File: portal/src/main/webapp/js/src/study-view/view/StudyViewInitCharts.js [hunk 1]
```diff
@@ -683,27 +686,26 @@ var StudyViewInitCharts = (function(){
...
+
+    function getRedrawService(exceptionIds){
         var _selectedCases = getSelectedCases().map(function(e){
             return e.CASE_ID;
         });
 
         var redrawService = {
             selectedCases: _selectedCases,
-            exceptionIds: exceptionIds
+            exceptionIds: exceptionIds==undefined?[]:exceptionIds
```
Comment (zhx828) [inline portal/src/main/webapp/js/src/study-view/view/StudyViewInitCharts.js]: I would recommend to use _.isUndefined or typeof exceptionIds === 'undefined'

File: portal/src/main/webapp/js/src/study-view/view/StudyViewInitTables.js [hunk 2]
```diff
@@ -427,44 +427,89 @@ var StudyViewInitTables = (function() {
...
+        if (exceptionIds.indexOf(worker.opts.tableId) === -1) {
+            worker.tableInstance.startLoading();
+        }
+
+    }
+
+    // find the worker for a tableId
+    function findWorker(tableId){
+        for(var i=0; i<workers.length; i++){
+            if (workers[i].opts.tableId == tableId) {
```
Comment (zhx828) [inline portal/src/main/webapp/js/src/study-view/view/StudyViewInitTables.js]: ===？

**Comment by zhx828** [comment]
@aderidder @pieterlukasse The code looks good to me. Could we make the cross dots a little bigger?

**Comment by pieterlukasse** [comment]
@zhx828 unfortunately the crosses also become thicker when made bigger...so maybe this will look even worse.

File: portal/src/main/webapp/js/src/study-view/view/StudyViewInitCharts.js (hunk 1 above)
Comment (aderidder) [inline portal/src/main/webapp/js/src/study-view/view/StudyViewInitCharts.js]: ok! fixed

File: portal/src/main/webapp/js/src/study-view/view/StudyViewInitTables.js (hunk 2 above)
Comment (aderidder) [inline portal/src/main/webapp/js/src/study-view/view/StudyViewInitTables.js]: are you against using === here?

**Comment by zhx828** [comment]
@pieterlukasse Then let's keep current size. Nice work!

File: portal/src/main/webapp/js/src/study-view/view/StudyViewInitTables.js (hunk 2 above)
Comment (zhx828) [inline portal/src/main/webapp/js/src/study-view/view/StudyViewInitTables.js]: I think you should use === in line 502 instead of ==.

Comment (aderidder) [inline portal/src/main/webapp/js/src/study-view/view/StudyViewInitTables.js]: fixed!

**Pull request merged**